search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt

```diff
@@ -129,21 +132,29 @@ class EntityEventListenerService(
     private fun handleAttributeUpdate(
         entityId: URI,
         expandedAttributeName: String,
+        datasetId: URI?,
         attributeValuesNode: JsonNode,
         updatedEntity: String,
         contexts: List<String>
     ) {
         // TODO add missing checks:
         //  - existence of temporal entity attribute
-        when (val extractedAttributeMetadata = toTemporalAttributeMedata(attributeValuesNode)) {
+        val compactedJsonLdEntity = addContextsToEntity(JsonUtils.deserializeObject(updatedEntity), contexts)
+        val attributeInstancePayload = extractAttributeInstanceFromCompactedEntity(
+            compactedJsonLdEntity,
+            compactTerm(expandedAttributeName, contexts),
+            datasetId
+        )
+        // Since ATTRIBUTE_UPDATE events payload may not contain the attribute type
+        if (attributeValuesNode.findValue("type") == null)
+        (attributeValuesNode as ObjectNode).put("type", attributeInstancePayload["type"] as String)
```

Comment:
missing indentation?

Reply:
No, it's the result of ktlintFormat task.

Reply:
yes but re-add the indentation, I don't get this.